src/wasm/kotlin/java/arcs/entity-class.kt

```diff
@@ -0,0 +1,117 @@
+// This is a reference for the entity classes generated by schema2wasm, based on the schema:
+//
+// particle Test
+//   data: reads {num: Number, txt: Text, lnk: URL, flg: Boolean, ref: &{val: Text}}
+
+package arcs
+
+class Test_Data_Ref() // Implementation omitted for brevity
+
+class Test_Data() : Entity<Test_Data>() {
+
+    var _numSet = false
+    var num = 0.0
+        get() = field
+        set(value) {
+            field = value
+            _numSet = true
+        }
+    var _txtSet = false
+    var txt = ""
+        get() = field
+        set(value) {
+            field = value
+            _txtSet = true
+        }
+    var _lnkSet = false
+    var lnk = ""
+        get() = field
+        set(value) {
+            field = value
+            _lnkSet = true
+        }
+    var _flgSet = false
+    var flg = false
+        get() = field
+        set(value) {
+            field = value
+            _flgSet = true
+        }
+
+    constructor(
+        num: Double,
+        txt: String,
+        lnk: String,
+        flg: Boolean
+    ) : this() {
+        this.num = num
+        this.txt = txt
+        this.lnk = lnk
+        this.flg = flg
+    }
+
+    override fun isSet(): Boolean {
+        return _numSet && _txtSet && _lnkSet && _flgSet
+    }
```

Comment:
This could probably be made into a property:

```kotlin
val isSet: Boolean
    get() = _numSet && _txtSet && _lnkSet && _flgSet
```

Reply:
Probably not; it's entirely possible for a schema to have a field called `isSet`. Now, if that still clashes with the method name we have a problem.